br/cmd/br/main_test.go

```diff
@@ -20,8 +20,39 @@ import (
 	"os"
 	"strings"
 	"testing"
+
+	"go.uber.org/goleak"
 )
 
+func TestMain(m *testing.M) {
+	skipLeakTest := false
+	var newArgs []string
+	for _, arg := range os.Args {
+		if arg == "--skip-goleak" {
+			skipLeakTest = true
+		} else {
+			newArgs = append(newArgs, arg)
+		}
+	}
+	os.Args = newArgs
+
+	if !skipLeakTest {
+		goleak.VerifyTestMain(m,
+			goleak.IgnoreCurrent(),
+			goleak.IgnoreTopFunction("github.com/pingcap/tidb/br/pkg/utils.StartExitSingleListener.func1"),
+			goleak.IgnoreTopFunction("github.com/pingcap/tidb/br/pkg/utils.StartDynamicPProfListener.func1"),
+			goleak.IgnoreTopFunction("gopkg.in/natefinch/lumberjack%2ev2.(*Logger).millRun"),
+			goleak.IgnoreTopFunction("go.etcd.io/etcd/client/v3.waitRetryBackoff"),
+			goleak.IgnoreTopFunction("net/http.(*persistConn).writeLoop"),
+			goleak.IgnoreTopFunction("internal/poll.runtime_pollWait"),
+			goleak.IgnoreTopFunction("google.golang.org/grpc.(*ClientConn).WaitForStateChange"),
```

Comment:
It seems this implies a open gRPC connection, is it safe to ignore this?

Reply:
This error shows in G07 and G08, and if you add a 1 min sleep before goleak, it doesn't show up. I believe there is a timeout, Therefore it should be harmless.